server/bin/pbench-tarball-stats

```diff
@@ -150,17 +157,29 @@ def main(options):
     print(f"{invalid_cnt:18n} bad tar balls, total count")
     print(f"{invalid_size:18n} bad tar balls, total size")
 
-    print(f"{len(good.keys()):18n} controllers with good tar balls")
+    print(
+        f"{len(good.keys()):18n} controllers with good tar balls (EC2 {ec2_cnt:n}, RH {rh_cnt:n})"
+    )
     print(f"{good_cnt:18n} good tar balls, total count")
     print(f"{good_size:18n} good tar balls, total size")
 
     for year, y_list in sorted(by_year.items()):
         count = len(y_list)
-        print(f"{year}: {count:7n}")
+        ec2_cnt = 0
```

Comment:
Would probably be better not to re-use the same variable name in the loops. I've tripped myself up before doing stuff like this because later I want to add some other output at the end using the original total, and then have to figure out why the number is different. ;-)

Of course another option would be to keep the "global" information as instance attributes, and do each output section in a method which shares the arrays and such but have their own local variables to keep them separate... which might make the whole thing easier to read.

Reply:
Refactored into a single function.